tls/s2n_server_hello.c

```diff
@@ -195,16 +205,31 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
     return 0;
 }
 
+static int s2n_hello_retry_required(struct s2n_connection *conn)
+{
+    notnull_check(conn);
+
+    return conn->handshake.requires_retry == 1; 
+}
+
 int s2n_server_hello_send(struct s2n_connection *conn)
 {
+    /* If a curve was not negotiated we need to retry */
+    if (s2n_hello_retry_required(conn) > 0) {
```

Comment:
I went back and forth on how to do a boolean comparison and I didn't see any standard way in the rest of the codebase. `if (s2n_hello_retry_required(conn))` doesn't work because a null `conn` value returns -1. I could always remove that notnull_check (its use is also sparse and I couldn't determine a standard).

My update will be removing the notnull_check actually. I'll change the function name to ` bool s2n_is_server_hello_required`. I think this would read better with that update.